tests/postgresql/usage/TimeZone.cpp

```diff
@@ -0,0 +1,159 @@
+ *
+ *   Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <vector>
+
+#include <sqlpp11/postgresql/postgresql.h>
+#include <sqlpp11/sqlpp11.h>
+
+#include "make_test_connection.h"
+#include "TabDateTime.h"
+
+namespace
+{
+  void save_regular (sqlpp::postgresql::connection& dbc, sqlpp::chrono::microsecond_point tp, std::chrono::microseconds tod, sqlpp::chrono::day_point dp)
+  {
+    model::TabDateTime tab {};
+    dbc(
+      update(tab)
+      .set(
+        tab.c_timepoint = tp,
+        tab.c_time = tod,
+        tab.c_day = dp
+      )
+      .unconditionally()
+    );
+  }
+
+  void save_prepared (sqlpp::postgresql::connection& dbc, sqlpp::chrono::microsecond_point tp, std::chrono::microseconds tod, sqlpp::chrono::day_point dp)
+  {
+    model::TabDateTime tab {};
+    auto prepared_update = dbc.prepare(
+      update(tab)
+      .set(
+        tab.c_timepoint = parameter(tab.c_timepoint),
+        tab.c_time = parameter(tab.c_time),
+        tab.c_day = parameter(tab.c_day)
+      )
+      .unconditionally()
+    );
+    prepared_update.params.c_timepoint = tp;
+    prepared_update.params.c_time = tod;
+    prepared_update.params.c_day = dp;
+    dbc(prepared_update);
+  }
+
+  template <typename L, typename R>
+  void require_equal(int line, const L& l, const R& r)
+  {
+    if (l != r)
+    {
+      std::cerr << line << ": ";
+      serialize(::sqlpp::wrap_operand_t<L>{l}, std::cerr);
+      std::cerr << " != ";
+      serialize(::sqlpp::wrap_operand_t<R>{r}, std::cerr);
+      throw std::runtime_error("Unexpected result");
+    }
+  }
+
+  void check_saved_values(sqlpp::postgresql::connection& dbc, sqlpp::chrono::microsecond_point tp, std::chrono::microseconds tod, sqlpp::chrono::day_point dp)
+  {
+    model::TabDateTime tab {};
+
+    // Check if the date/time values are saved using the UTC time zone
+    const auto &rows_1 = dbc(
+      select(
+        sqlpp::verbatim<sqlpp::integer>("floor(extract(epoch from c_timepoint)*1000000)::int8").as(sqlpp::alias::a),
```

Comment:
I admit, I have a hard time understanding what is selected here. Can you add a comment?

Reply:
Indeed the test seems to be more cryptic than the main code :-)
I added a few extra comments that hopefully make the test code clearer.
So please let me know if it needs additional clarification.

Reply:
I think this is clear enough now. Thanks for adding the comments.